### DISCUSSION TITLE
Cropper converting JPG to PNG

### DISCUSSION THREAD

**Discussion by wsdnathan** [orchidsoftware/platform#1702] [Q&A]
If I use `Cropper::make('picture');` and upload a JPG, the cropped image gets saved as a very large filesize PNG. Is there any way to preserve the original filetype of JPG so that I do not wind up with extremely large sized photos? For example a 100KB JPG cropped results in a 1MB PNG.

**Comment by czernika** [comment]
@tabuna what if in `cropper_controller.js` get file type (we can do it on upload)

```js
this.fileType = event.target.files[0].type
```

and use it in canvas `toBlob` as [second argument](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob#type)?

```js
this.cropper.getCroppedCanvas({
            // ...
        }).toBlob((blob) => {
            // ...
        }, this.fileType); // here
```

Also maybe it is worth to make `Cropper` field method something like `keepOriginalType(bool $keep = false)` - if it is true, image will preserve its extension otherwise converted to png (default current behaviour)

**Reply by tabuna** [reply]
Yes, I think it makes sense. Will you do PR?

**Reply by czernika** [reply]
[Done](https://github.com/orchidsoftware/platform/pull/2779). Changed logic little bit because when we call method named `keepOriginalType` we are assuming it will keep it without passing extra true param. However default value if method was not called is still `false` so images will be converted to PNG. In a new major release I think it is worth it to change to `true` - for now let's not break things
